### DISCUSSION TITLE
Next.js Documentation pdf book

### DISCUSSION THREAD

**Discussion by Mahmoud11350** [vercel/next.js#35643] [Help]
Is there way to download next.js documentation book . As I won't be able to open the internet for a while .

**Comment by alex-semenyuk** [comment]
If you mean nextjs docs https://nextjs.org/docs then you can find all content at github repo https://github.com/vercel/next.js/tree/canary/docs. So you clone this repo and use info offline.

**Reply by WeAsAn** [reply]
What about https://beta.nextjs.org/docs?

**Comment by uni-parse** [comment]
thanks @alex-semenyuk, that really helpful !!
but it will be nice if Vercel open source <a href='nextjs.org'>nextjs.org</a> so we can clone it & run it locally !!
I use <a href='react.dev'>react.dev</a> locally by cloning its <a href='https://github.com/reactjs/react.dev'>repo</a>,
I also use <a href='https://javascript.info/'>javascript.info</a> locally by cloning its <a href='https://github.com/javascript-tutorial/server'>repo</a>,
I also use <a href='https://tailwindcss.com/'>tailwindcss.com</a> locally by cloning its <a href='https://github.com/tailwindlabs/tailwindcss.com'>repo</a>
![bitmoji](https://sdk.bitmoji.com/render/panel/525c1f1a-a423-4aa8-9b81-f1088757d528-74f285c6-c7cc-406e-80df-2ceebc8ce9db-v1.png?transparent=1&palette=1&width=246)

**Reply by aftababu** [reply]
@TheUniParse  bro ,how do you do that ? that gave me a huge error.

**Reply by emailjohnthomascaballero** [reply]
can you provide exact thing to do to make it work locally?

**Comment by nerder** [comment]
As a side note, it will be useful to have the doc exported in PDF to be able to upload it to ChatGPT. Since is capped to 2021 is not easy to find good insights for how to properly use the new `app` directory for instance.

**Reply by Philnue** [reply]
You solved it ?

**Reply by Nikhi-l** [reply]
does uploading the documentation in the new gpt4 bot helps?

**Reply by veerbal1** [reply]
Yeah, we need it

**Reply by q2dynamix-com** [reply]
You can provide the link to the documentation and it seems that your GPT will be trained with it.

**Reply by Nikhi-l** [reply]
GPT-4's training includes data only up to April 2023. For tasks involving newer technologies, like Next.js 14, i was looking to provide a PDF or document for reference.

**Reply by TatianKuzmina** [reply]
[docs-next_compressed.pdf](https://github.com/vercel/next.js/files/13598859/docs-next_compressed.pdf)

**Reply by tomer183** [reply]
Hey @TatianKuzmina thanks!, this file is for version 14 ?

**Comment by Riyooo** [comment]
I also wanted to export the documentation for training a ChatGPT bot on updated data.
Here's my last v14 pdf in case it helps: [Next.js_v14_Documentation.pdf](https://github.com/vercel/next.js/files/13780346/Next.js_v14_Documentation.pdf)

**Reply by tomer183** [reply]
thanks

**Reply by dr-dark-flames** [reply]
how do i train a chatgpt bot on such data? does it need the paid version? and if paid do i upload the pdf and thats it? if not paid, then how or where do we train a bot on such data? 
Thank you so much for the docs btw <3

**Reply by Riyooo** [reply]
Happy to help :) I effectively think that it's a paid feature, giving the ability to explore GPTs created by others or create your own. You are then presented with a form where you give your custom GPT a name and instructions on what you want it to do and how you want it to interact with you. You can also import "knowledge", and that's where I uploaded the PDF. Initially I tried asking the GPT to always look at the latest documentation on next.js official website before answering me but it seemed to not respect these instructions. I then tried feeding it with the documentation as a pdf directly and it works far better.

**Reply by dr-dark-flames** [reply]
much appreciated

**Reply by Romney81** [reply]
@Riyooo Out of curiosity, how did you achieve this? Did you put this pdf together manually?

**Reply by Nikhi-l** [reply]
https://chat.openai.com/g/g-tQJVsUrAT-next-js-helper for folks already having the plus plan

**Reply by alexsoyes** [reply]
@Riyooo Amaaaaaazing thank you so much! How did you generate the PDF in the first place? From GitHub with a tool?

**Reply by tomer183** [reply]
https://chat.openai.com/g/g-NoR8ssMVq-next-js-14-expert is working just fine.

**Reply by sahilatahar** [reply]
Thanks

**Reply by DelicTide** [reply]
This is SO great! Thank you soo much!!

**Reply by emailjohnthomascaballero** [reply]
thnx for this

**Reply by morninng** [reply]
how did you make it?
i want to get the same with a latest document

**Reply by Riyooo** [reply]
@alexsoyes @Romney81 @morninng I made a python script for that.
In case it helps I uploaded it here : [https://github.com/Riyooo/Docs-Exporter](https://github.com/Riyooo/Docs-Exporter)
The latest documentation is also in the releases section and I will try to update it regularly.
[Next.js_Docs_v15.0.0_2024-05-17.pdf](https://github.com/Riyooo/Docs-Exporter/releases/download/v15.0.0_2024-05-17/Next.js_Docs_v15.0.0_2024-05-17.pdf)
The GPT for which I use this documentation: https://chatgpt.com/g/g-fKeb064KB-gpt-next

**Reply by dr-dark-flames** [reply]
W Riyooo

**Reply by Sawankumarjha-1** [reply]
Thanks buddy

**Reply by getong** [reply]
Nextjs 15.0.1 is released now, will it be updated?

**Reply by Riyooo** [reply]
Done, thanks for the reminder

**Reply by Ibrahim-Sarhan** [reply]
Thank you very much @Riyooo

**Reply by patryk672** [reply]
@Riyooo maybe you know how to do the same for https://astro.build/?

**Reply by pacnpal** [reply]
> @Riyooo maybe you know how to do the same for [astro.build](https://astro.build/)?

I forked @Riyooo and did just that: https://github.com/PacFactory/Docs-Exporter-Astro/releases/

**Reply by champi2** [reply]
> Je voulais aussi exporter la documentation pour la formation d'un robot ChatGPT sur les données mises à jour. Voici mon dernier v14 pdf s'il aide : [Next.js-v14-Documentation.pdf](https://github.com/vercel/next.js/files/13780346/Next.js_v14_Documentation.pdf)

please you have the last documentation next js 15.3.0 PDF

**Reply by Riyooo** [reply]
@champi2  just updated ! https://github.com/Riyooo/Docs-Exporter/releases

**Reply by getong** [reply]
@Riyooo 
The pdf code is something wrong, might need to fix the error
```
export default function Page() {
return &lt;h1 className="text-3xl font-bold underline"&gt;Hello, Next.js!&lt;/h1&gt;
}
```
in page 40
The `&lt;` `&gt;` is wrong text in pdf file

**Comment by shahriarshafin** [comment]
Hey folks! Just built my NextJS Copilot with Google AI Studio. It's working insanely well! 🚀 thanks @Riyooo for the pdf

<img width="474" alt="Screenshot 2024-05-16 at 4 46 25 PM" src="https://github.com/vercel/next.js/assets/32214710/5edad229-7e16-4836-aa7c-5b21d2e08e5f">

**Reply by dr-dark-flames** [reply]
what is ur process to making one?? do u do anything else other than uploading the pdf? what is the system prompt?

**Reply by shahriarshafin** [reply]
i just uploaded the pdf with some basic prompt. that's it.

**Reply by tomer183** [reply]
can you write the creation process?

**Reply by shahriarshafin** [reply]
Hi @tomer183, thanks for your interest
Here’s how you can create your own:
- goto https://aistudio.google.com/app/prompts/new_chat
- upload the pdf or other files from your drive
- copy / paste this as System Instructions

> Objective: Solve coding problems specifically adhering to the Next.js 14 structure.
Input:
Next.js PDF Documentation: A single PDF document containing comprehensive information on the Next.js framework.
Output:
Solution Code: A code snippet or complete solution that aligns with the Next.js structure and best practices as described in the PDF documentation by providing clear and concise solutions.
Explanation: A brief explanation of how the solution addresses the problem, with references to relevant sections of the PDF documentation.
- that's it!

**Reply by tomer183** [reply]
Thanks

**Reply by internalG** [reply]
Be careful with that pdf. It mixed app with pages router. Your AI answer could be wrong.

**Reply by shahriarshafin** [reply]
now https://v0.dev/ do the job for me!
